### ISSUE TITLE
1.15 follow-up for new menu click/touch activate description text

### ISSUE THREAD

**Issue by jenlampton** [backdrop/backdrop-issues#4253]
I also find this change very useful, as it exposes an already existing feature of the 3rd party library we use for the drop-down menus 👍 

The help text for the new checkbox was initially this:

> When checked, the menu will expand when clicked rather than when hovered.

Because that was simply repeating the label of the checkbox, I suggested this alternative:

> This option may work better on mobile, where there is no cursor, and menu expand/collapse relies on touch events.

[edit] ...which now, after @herbdool's suggestion bellow, I would like to change to this:

> This option may work better on devices where there is no cursor, and menu expand/collapse relies on touch events instead.

And we finally ended up with this:

> Menus that don't appear or disappear on hover can be better for usability in some cases.

cc'ing @jenlampton, as she usually has better words for things 😄

**Comment by jenlampton** [comment]
My comment from the other issue...

> This option may work better on mobile, where there is no cursor, and menu expand/collapse relies on touch events.

The current core menu already uses click/touch on mobile, where there is no cursor. This setting is to make click/touch replace the hover setting on **desktop**. This new description text is misleading. How about we remove the description text entirely if it is identical to the label?

> I wanted to note that this won't affect small mobile screens since the mobile menu already requires clicks.

Ah, yes. What @herbdool said :)

> This option may work better on devices where there is no cursor, and menu expand/collapse relies on touch events instead.

But the core menus already work on MOST devices where there is no cursor. I still don't think this text is helping...

> Menus that don't appear or disappear on hover can be better for usability in some cases.

Are you also recommending this text for the description for the checkbox? I'm not sure what this text has to do with the setting...

**Comment by jenlampton** [comment]
Let's delete the description text entirely and change the label to 

`Always use click to open sub-menus (instead of hover on large-screen devices)`

**Comment by quicksketch** [comment]
Although radio buttons take up more space, would this be more clear?

```
Menu Drop-downs
(*) Open on mouse hover
( ) Open on mouse click
This setting applied to devices with a mouse cursor. Touch-screen devices always open on touch.
```

**Comment by olafgrabienski** [comment]
I like the radio buttons! There's a small typo, btw: "This setting applie**d** ...". However, in my opinion we don't need this sentence in the help text, it's perfectly fine without, isn't it?

```
Menu Drop-downs
(*) Open on mouse hover
( ) Open on mouse click
Touch-screen devices always open on touch.
```

**Comment by laryn** [comment]
I like simplifying things -- this all looks good to me.

>> Menus that don't appear or disappear on hover can be better for usability in some cases.
> Are you also recommending this text for the description for the checkbox? I'm not sure what this text has to do with the setting...

Probably unnecessary to include this in the description -- it's just saying that click menus that stay open can be more accessible/better usability for certain cases (e.g. primarily older audience, etc.).

**Comment by jenlampton** [comment]
The current suggestion doesn't reflect what the setting actually changes. 

The default touch/hover change is caused by device screen size, not whether they are Touch-screen devices or not. This is the problem we are trying to solve. Large iPads, for example, use the hover effect even though they are Touch-screen devices.

Something like the following would be more correct:

Menu Drop-down behavior for devices with large screens:
(*) Open on mouse hover 
( ) Open on mouse click 
Devices with smaller screens always open on touch.

**Comment by olafgrabienski** [comment]
> The default touch/hover change is caused by device screen size, not whether they are touch-screen devices or not.

👍 

> I've created a PR, do you think this is too wordy? Suggestions?

Okay, let's try to use fewer words:

```
Dropdown behavior on large screens:
(*) Open with mouse hover
( ) Open with mouse click or touch
Dropdowns on smaller screens will always open on touch.
```

(Note: I've used `Dropdown`  instead of `Drop-down` as the *Menu style* in the configuration dialog is called `Dropdown menu`.)

**Comment by jenlampton** [comment]
Better!  thanks @olafgrabienski :) PR updated.

**Comment by olafgrabienski** [comment]
> PR updated

Looks good to me, only one thing: I think we don't use colons for the heading of a setting.

`Dropdown behavior on large screens:` should be `Dropdown behavior on large screens`

(Sorry @jenlampton - I forgot to remove the colon when I copied your former suggestion to adapt it.)

**Comment by jenlampton** [comment]
@quicksketch thinks the touch/hover effect is triggered by JavaScript and not by the same media query that changes the menu from desktop to mobile visually, so we might need to do some more testing to find out exactly how this thing works before we can add useful description text :)

**Comment by jenlampton** [comment]
Adjusting milestone, this can be done at any time.

**Comment by olafgrabienski** [comment]
>> The default touch/hover change is caused by device screen size, not whether they are touch-screen devices or not. (...) Large iPads, for example, currently use the hover effect even though they are Touch-screen devices.

> @quicksketch thinks the touch/hover effect is triggered by JavaScript and not by the same media query that changes the menu from desktop to mobile visually

@jenlampton I've tested the menu behavior on a Backdrop site with the default (Open with mouse hover) setting:

- On a large iPad (Pro 11'', resolution equivalent to 1112 x 834px), sub-menus open on touch.
- On a small notebook browser window (768px), sub-menus open on hover.
- On even smaller notebook browser windows (less than 768px), the whole menu gets replaced by the 'Hamburger' menu.

So, if I don't miss something, the sentence "Touch-screen devices always open on touch" was actually correct. Based on that assumption, here's my suggestion:

```
Dropdown behavior on large screens
(*) Open with mouse hover
( ) Open with mouse click or touch
Touch-screen devices always open on touch.

**Comment by klonos** [comment]
Before:

![Screen Shot 2020-08-17 at 3 41 42 pm](https://user-images.githubusercontent.com/2423362/90397503-9f496d00-e0a0-11ea-913b-8c1f2c37832c.png)

After:

![Screen Shot 2020-08-17 at 3 41 59 pm](https://user-images.githubusercontent.com/2423362/90397522-a53f4e00-e0a0-11ea-92f0-2c4646e925f3.png)

**Comment by klonos** [comment]
@jenlampton I've left a tiny nitpicking suggestion in the PR. Other than that, this is RTBC.

**Comment by docwilmot** [comment]
Attempting a summary to get this clear in my head:
**touchscreens**
- there is no hover, I've Googled, and @klonos confirmed on Zulip; hover is not a thing on touchscreen
- screen size does not matter, they all have no hover
- menus always open on click

** NON-touchscreens**
- regardless of the size, if there is a mouse, it will obey the radio option to hover vs click

THerefore, mentioning screen size is irrelevant. 

The best option for wording this issue then is @olafgrabienski in https://github.com/backdrop/backdrop-issues/issues/4253#issuecomment-570175116

**Comment by klonos** [comment]
One correction:

> ...hover is not a thing on touchscreen

Not a thing on touchscreen**-only**. There exist devices/laptops that are touch-/pen-enabled, yet they also allow for mice to be connected and used.

Either way, we should test this and adapt the wording a bit.

**Comment by docwilmot** [comment]
But thats not what we're talking about here. If youre using it with a mouse youre not using it as a touchscreen. Regardless, the behavior will be according to the radio option, so the size issue still is irrelevant.

**Comment by jenlampton** [comment]
We've determined that size is irrelevant. We've determined the device being a touchscreen or not-touchscreen is also irrelevant. It sounds like the only deciding factor is whether you are using a finger or a cursor to trigger the menu? 

How about something like this:

```
Dropdown menu behavior with cursor:
* Open when cursor hovers
* Open when cursor clicks
```

(I think we should avoid 'mouse' because many people use trackpads instead of mice)

**Comment by laryn** [comment]
I just tested this on a touchscreen laptop. With the settings on the menu block not set to use the 'click' functionality (just default hover) I found:

 - trackpad works with hover, as expected (hovering opens the submenu and a click on a main nav item automatically opens the target)
 - touching a main menu item on the touchscreen functions as a click/touch open (ie. rather than automatically directing to the menu target it opens the menu and waits for another touch)

**Comment by docwilmot** [comment]
> Dropdown behavior
Open with hover or touch
Open with click or touch

I think that works

**Comment by jenlampton** [comment]
I've updated the PR to use: **Dropdown menu trigger** as the label. I felt that `behavior` was too vague. I also moved the part of the option that was different to the end of the line, so that as people are scanning it will be easier to pick out where the two options differ (and where they are the same).

* Open menu with touch or hover
* Open menu with touch or click

**Comment by jenlampton** [comment]
I dislike behaviour because that could mean anything about how the menu
behaves: how fast or slow it opens, if it slides open or blinks, etc. We
may have other behaviors we want to add settings for. This label should
match this specific behavior. Can you think of a better word than "trigger"
(rather than going back to the word that's problematic) that's specific to
what opens the submenu?

I want to be sure people will know what this setting does. Saying it's a
"behavior that opens" isn't very descriptive (especially if you don't think
of a menu as "opening", which I expect is that case in some languages). If
we provide some context as to what is opening what, it would be more
helpful.

*brainstorm*
Behavior that opens
Action that opens child menu
Event that reveals inner menu
Interaction that displays more links
???

**Comment by klonos** [comment]
Feedback:
- I understand the vagueness of "behavior" 👍🏼 
- I don't like the word trigger, but it is necessary in various things in IT, so I avoid it if possible.
- The label starts with the words "dropdown" and "menu", whereas we should try to emphasize the verb we choose for the action (be it "trigger" or "open" or "reveal" or "display")
- We are repeating "Open menu with" for both options, which could be used as a label.

How about:
```
Open dropdown menu with
- Touch or hover
- Touch or click
```
(not sure whether "with" or "on" or "upon" is more appropriate)

...or:
```
Open dropdown menu when
- Touched or hovered
- Touched or clicked
```

**Comment by olafgrabienski** [comment]
> Can you think of a better word than "trigger" (...) that's specific to what opens the submenu?

Let's see! Maybe including one of the terms "to open" or "submenu" makes it even more clear:

`Dropdown submenu trigger` is very clear. Sounds however quite technical and a bit awkward to me. Advantage: the options can be shorter.

```
Dropdown submenu trigger
- Open with touch or hover
- Open with touch or click
```

`Dropdown opening behavior` might be technically less clear, but it sounds very understandable if we use the term "submenu" in the options:

```
Dropdown opening behavior
- Open submenu with touch or hover
- Open submenu with touch or click
```

**Comment by klonos** [comment]
🤔 ...well, _technically_ you are not "opening the submenu", rather than "the submenus are revealed when you open the parent menu item".

In this specific context, the menu has one behavior and that is to open and close (or "expand" and "collapse" if you like). What we want to describe here though is the various available options that cause it ("triggers"). The triggers are: touch/hover/click.

When it comes to the various verbs being used to describe actions:
- the parent menu is opened/expanded and closed/collapsed
- the submenus are shown/revealed and hidden

So my previous suggestion would be either:
```
Open dropdown menu when it is
- Touched or hovered
- Touched or clicked
```

...or:
Show submenus when the parent menu is
- Touched or hovered
- Touched or clicked
```

**Comment by olafgrabienski** [comment]
Thanks @klonos. I'm not sure if I agree to emphasize the verb in the label here. The approach sounds generally elegant, but I think a sentence shouldn't be split up in two parts (label and option), so that the option isn't descriptive without the label. To avoid issues, e.g. re translations, I'd prefer to stick with a traditional label like `Dropdown opening behavior` and more complete options, e.g. my suggestion from above but using "show" instead of "open".

```
Dropdown opening behavior
- Show submenu with touch or hover
- Show submenu with touch or click
```

**Comment by klonos** [comment]
> ...I think a sentence shouldn't be split up in two parts (label and option), so that the option isn't descriptive without the label. To avoid issues, e.g. re translations...

Right. I haven't thought about it that way.

**Comment by jenlampton** [comment]
I came here to review/test this one for 1.33.1, but it is my own work :/  
I was able to help with a rebase though ;)

**Comment by quicksketch** [comment]
I did not see any substantial issues with https://github.com/backdrop/backdrop/pull/3034. Some syntax could maybe be shortened, but after kicking this forward for 40+ releases I'd like to just get it in since it works just fine and we all agree.

I merged https://github.com/backdrop/backdrop/pull/3034 into 1.x and 1.33.x. Thank you @jenlampton, @olafgrabienski, @klonos, @docwilmot, and @laryn for collaborating on this one!
